test/tests/functional/pbs_soft_walltime.py

```diff
@@ -852,46 +831,64 @@ def test_walltime_default(self):
+        # try:
+        #    J = Job(TEST_USER, attrs={'Resource_List.walltime': 10})
+        #    jid1 = self.server.submit(J)
+        # except PtlSubmitError as e:
+        #    self.assertTrue("illegal attribute or resource value" in e.msg[0])
+        # self.assertEqual(jid1, None)
 
-    def test_soft_suspension(self):
+    def test_soft_held(self):
         """
-        Test that if job is suspended soft_walltime will not get extended
+        Test that if job is held soft_walltime will not get extended
         """
-
-        J = Job(TEST_USER, {'Resource_List.walltime': 300})
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': '100'})
         jid = self.server.submit(J)
         self.server.alterjob(jid, {'Resource_List.soft_walltime': 7})
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
 
-        self.server.sigjob(jid, signal='suspend')
-        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+        now = int(time.time())
 
+        # Sleep so long that soft_walltime get extended once
         time.sleep(7)
-        self.server.expect(JOB, {'Resource_List.soft_walltime': 7}, id=jid)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:14|14')}, id=jid)
 
-        self.server.sigjob(jid, signal='resume')
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.holdjob(jid, 'u')
+        self.server.rerunjob(jid)
+        self.server.expect(JOB, {'job_state': 'H'}, id=jid)
 
-        time.sleep(5)
-        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        # sleep so long to verify that soft_walltime don't change
+        # while job is held
+        time.sleep(10)
         self.server.expect(JOB, {'estimated.soft_walltime':
                                  (MATCH_RE, '00:00:14|14')}, id=jid)
 
+        # release the job and look for the soft_walltime again
+        self.server.rlsjob(jid, 'u')
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'job_state': 'R', 'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:14|14')}, id=jid)
+
+        # Wait for some more time and verify that soft_walltime
+        # extending again
+        time.sleep(7)
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:21|21')}, id=jid)
+
     def test_soft_less_cput(self):
         """
         Test that soft_walltime has no impact on cput enforcement limit
         """
 
-        script = []
-        script += ['i=0;\n']
-        script += ['while [ $i -ne 0 ] || sleep 0.125;\n']
-        script += ['do i=$(((i+1) % 10000 ));\n']
-        script += ['done\n']
+        script = """
+i=0;
+while [ $i -ne 0 ] || sleep 0.125;
+do i=$(((i+1) % 10000 ));
+done
```

Comment:
I find your while header confusing.  Loop headers shouldn't do anything, they should just test.  What you have done is basically the same as having the sleep as the first thing in the while loop.  Why not put it in the loop body itself?